### PR TITLE
Project5

### PR DESCRIPTION
**Pull request by amrutha-chil**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,35 +1,37 @@
...
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    <div>
```
Comment (sirKevlar) [inline src/App.jsx:21]: I think we can be more semantic with our choice of HTML tag here... Usually divs are used purely for styling.

File: src/App.jsx [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
+import "./App.css";
+import { Navbar } from "./components/Navbar";
+import axios from "axios";
+import { Articles } from "./components/articles";
+import { Article } from "./components/Article";
+import { Route, Routes } from "react-router-dom";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [articles, setArticles] = useState([]);
```
Comment (sirKevlar) [inline src/App.jsx:10]: As our app has a useEffect making an api call, loading state should start off as true

File: src/components/Article.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from "react";
+import { getArticles, getArticlesById } from "../api";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+
+export const Article = () => {
+  const params = useParams();
+  const [article, setArticle] = useState(null);
```
Comment (sirKevlar) [inline src/components/Article.jsx:8]: This loading state should be initialised as true

File: src/components/Article.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            alt={`cover of the ${article.title} book`}
+          />
+          <div className="article-items">Topic - {article.topic}</div>
+          <div className="article-items">
+            Comment Count - {article.comment_count}
+          </div>
+          <div className="article-btns">
+            <button>Votes - {article.votes}</button>
+            <button>Comment</button>
+          </div>
```
Comment (NRMSMITH) [inline src/components/Article.jsx:44]: You've not displayed the the body of the article - so no one can read it!

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Just rendering the body! :D

File: src/App.jsx [hunk 5]
```diff
@@ -1,35 +1,37 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useEffect, useState } from "react";
+import "./App.css";
+import { Navbar } from "./components/Navbar";
+import axios from "axios";
+import { Articles } from "./components/articles";
```
Comment (mezzDavies) [inline src/App.jsx:5]: This needs to be a capital "A" for /articles

File: src/components/Navbar.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+export const Navbar = () => {
+  return (
+    <div className="navbar">
```
Comment (mezzDavies) [inline src/components/Navbar.jsx:6]: There is a better tag to use here in a **nav** section 😉

File: src/App.jsx [hunk 7]
```diff
@@ -1,35 +1,37 @@
...
-  )
+    <div>
+      {loading ? (
+        "Loading Page..."
+      ) : (
+        <main>
+          <Navbar />
+          <Routes>
+            <Route path="/" element={<Articles articles={articles} />}></Route>
+            <Route path="/article/:article_id" element={<Article />}></Route>
```
Comment (mezzDavies) [inline src/App.jsx:29]: Ideally the route should be "/articles/:article_id" - note the "s" on articles - this is then RESTful 👍

**Review by mezzDavies — APPROVED** [review]
Just a few small things to think about in the comments below but this is approved. You could consider extracting your loading into a separate component. Also adding the timestapm to the article (in a human readable format.
Finally [this](https://www.refactoringui.com/previews/labels-are-a-last-resort) might be of interest regarding the use of labels eg Topic or Author

**Pull request merged**